boards/Makefile.common

```diff
@@ -47,12 +48,24 @@ ifneq ($(shell RUSTUP_TOOLCHAIN=$(RUSTUP_TOOLCHAIN) rustup component list | grep
   $(shell RUSTUP_TOOLCHAIN=$(RUSTUP_TOOLCHAIN) rustup component add rust-src)
 endif
 
-# Check that xargo is installed
-ifneq ($(strip $(firstword $(shell $(XARGO) --version 2>&1))), xargo)
+# Check that xargo is installed and up-to-date
+XARGO_CALL := $(shell $(XARGO) --version 2>&1)
+XARGO_NAME := $(strip $(firstword $(XARGO_CALL)))
+XARGO_VERSION := $(strip $(word 2, $(XARGO_CALL)))
+ifneq ($(XARGO_NAME), xargo)
   $(warning Required tool `xargo` not installed.)
   $(warning Running `cargo install xargo` in 3 seconds (ctrl-c to cancel))
   $(shell sleep 3s)
   $(shell cargo install xargo)
+else ifeq ($(shell expr $(XARGO_VERSION) \< $(MINIMUM_XARGO_VERSION)), 1)
+  $(warning Required tool `xargo` is out-of-date.)
+  $(warning Running `cargo install cargo-update` and `cargo install-update xargo` in 3 seconds (ctrl-c to cancel))
+  $(shell sleep 3s)
+  $(shell cargo install cargo-update)
+  XARGO_UPDATE_RESULT := $(shell cargo install-update xargo))
+  $(info $(XARGO_UPDATE_RESULT))
+else
+  $(info Found `xargo` $(XARGO_VERSION))
```

Comment:
Move this down into the verbose block below, re-running xargo version to pick the results of any install we may have done